test/autorest/lrogroup/lros_test.go

```diff
@@ -96,421 +96,532 @@ func TestLROBeginDelete202NoRetry204(t *testing.T) {
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	rt, err := poller.ResumeToken()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	poller, err = op.ResumeLrOSDelete202Retry200Poller(rt)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	for poller.Poll(context.Background()) {
-// 		time.Sleep(200 * time.Millisecond)
-// 	}
-// 	resp, err := poller.Response()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp.RawResponse, 200)
-// 	resp, err = poller.Wait(context.Background(), time.Duration(1)*time.Second)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp.RawResponse, 200)
-// 	_, err = poller.ResumeToken()
-// 	if err == nil {
-// 		t.Fatal("did not receive an error but was expecting one")
-// 	}
-// }
-
-// func TestLROBeginDelete204Succeeded(t *testing.T) {
-// 	op := getLROSOperations(t)
-// 	poller, err := op.BeginDelete204Succeeded(context.Background())
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	_, err = poller.ResumeToken()
-// 	if err == nil {
-// 		t.Fatal("expected an error but did not receive one")
-// 	}
-// 	for poller.Poll(context.Background()) {
-// 		time.Sleep(200 * time.Millisecond)
-// 	}
-// 	resp, err := poller.Response()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp, 204)
-// 	resp, err = poller.Wait(context.Background(), time.Duration(1)*time.Second)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp, 204)
-// 	_, err = poller.ResumeToken()
-// 	if err == nil {
-// 		t.Fatal("did not receive an error but was expecting one")
-// 	}
-// }
-
-// func TestLROBeginDeleteAsyncNoHeaderInRetry(t *testing.T) {
-// 	op := getLROSOperations(t)
-// 	poller, err := op.BeginDeleteAsyncNoHeaderInRetry(context.Background())
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	rt, err := poller.ResumeToken()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	poller, err = op.ResumeLrOSDeleteAsyncNoHeaderInRetryPoller(rt)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	for poller.Poll(context.Background()) {
-// 		time.Sleep(200 * time.Millisecond)
-// 	}
-// 	resp, err := poller.Response()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp.RawResponse, 200)
-// 	resp, err = poller.Wait(context.Background(), time.Duration(1)*time.Second)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp.RawResponse, 200)
-// 	_, err = poller.ResumeToken()
-// 	if err == nil {
-// 		t.Fatal("did not receive an error but was expecting one")
-// 	}
-// }
+func TestLROBeginDelete202Retry200(t *testing.T) {
+	op := getLROSOperations(t)
+	resp, err := op.BeginDelete202Retry200(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller := resp.Poller
+	rt, err := poller.ResumeToken()
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller, err = op.ResumeDelete202Retry200(rt)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for !poller.Done() {
```

Comment:
Since PollUntilDone() uses this machinery internally I don't think it's necessary to duplicate the test.  Just call PollUntilDone().

Reply:
Done